### ISSUE TITLE
audit-engine prints "set laws.minLawsChecked against N" from --mode=full, but the hook it installs runs --mode=pre-commit, which selects fewer — following the advice fails every commit

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#141]
**Version:** 7.23.0
**Law:** not a law — `AuditEngine` in `dist/core/auditing/audit-engine.js`, the liveness-floor check at :117 and the advice printed at :425

Kind: the tool's own printed advice, followed exactly, disarms the tool's own hook.

## What happens

`--mode=full` ends with a line telling you what to set the floor to:

```
Total Laws Checked: 115
  of which count for the liveness floor: 113 (+2 always-on meta-gates — set laws.minLawsChecked against 113)
✗ Failed: 0
✅ CONSTITUTIONAL COMPLIANCE: PASSED
```

Setting `laws.minLawsChecked: 113` as instructed makes every commit fail:

```
$ git commit -F msg.txt
❌ Only 110 laws selected — below the liveness floor (laws.minLawsChecked: 113) — refusing to report compliance.
   The law selection collapsed. Check laws.enabled / laws.paretoMode / laws.notApplicable and the detected project type. A shrunken audit is a disarmed gate, never PASSED.
husky - pre-commit script failed (code 1)
```

The hook `roc init` installs runs a narrower mode:

```
$ grep ruleofcode .husky/pre-commit
npx ruleofcode audit --mode=pre-commit --staged
```

`--mode=pre-commit` selects 110 substantive laws where `--mode=full` selects 113. One setting governs both, so advice derived from the wider mode is guaranteed to break the narrower one.

## Reproduction

```
$ npx ruleofcode audit --mode=full --config ruleofcode.config.json | tail -5
  of which count for the liveness floor: 113 (... set laws.minLawsChecked against 113)

$ node -e "const c=require('./ruleofcode.config.json'); c.laws.minLawsChecked=113;
  require('fs').writeFileSync('ruleofcode.config.json', JSON.stringify(c,null,2))"

$ npx ruleofcode audit --mode=pre-commit --staged; echo $?
❌ Only 110 laws selected — below the liveness floor (laws.minLawsChecked: 113)
1
```

The count is stable, not staged-file-dependent — measured with six files staged and with one, 110 both times. So this is not flakiness; the two modes simply select different amounts and the advice names the larger.

## Where

`dist/core/auditing/audit-engine.js:425` prints the advice using the current run's own count:

```js
console.log(`${chalk.gray('  of which count for the liveness floor:')} ${chalk.white(substantive)} ${chalk.gray(`(+${auditResult.totalLaws - substantive} always-on meta-gates — set laws.minLawsChecked against ${substantive})`)}`);
```

`:117` enforces it against whatever mode is running:

```js
const laws = this.getLawsToAudit();
const substantive = laws.filter(law => !law.alwaysEnabled);
const floor = this.config.laws.minLawsChecked ?? 0;
if (substantive.length === 0 || substantive.length < floor) { ... }
```

And `dist/config/types.d.ts:46` confirms there is one knob, not one per mode:

```ts
minLawsChecked?: number;
```

## What it cost us

A failed commit, then a worse second move. The obvious repair is to lower the number until the commit passes — I set 107, which passed, and only afterwards noticed that 107 leaves three laws of slack. A floor with slack is precisely the hole the setting exists to close; the config's own note calls it "a floor, not a target". The right value is the tightest real selection, 110, which I then proved load-bearing by parking one extra law:

```
❌ Only 109 laws selected — below the liveness floor (laws.minLawsChecked: 110)
```

The failure message is good — it names the floor, the count, and where to look. The advice line is the part that misleads, because it is printed by the mode least likely to be the binding one.

## Suggested fix

1. Make the advice mode-aware — either name the mode it applies to, or print the narrowest selection the installed hooks will produce:

   ```
   set laws.minLawsChecked against 110 (the narrowest installed mode, --mode=pre-commit;
   this full run selected 113)
   ```

2. Or let the floor be per mode, so each can be set to its own tightest value:

   ```jsonc
   "minLawsChecked": { "full": 113, "pre-commit": 110 }
   ```

(1) is the smaller change and removes the trap on its own.

## What I did NOT verify

- Whether `--mode=pre-push` selects a third number. Our pre-push hook runs `npm run verify`, which calls `--mode=full`, so it did not arise here.
- Whether the gap between the two modes is stable across projects, or specific to how many laws this Angular/Nx workspace parks. I measured only this repo: 113 full, 110 pre-commit, with 17 laws in `notApplicable`.
- Whether `paretoMode` changes the relationship. We do not use it.
